### PR TITLE
Fix `TextAlignmentExtensions`

### LINKED ISSUE #97: [Bug] TextCenterVertical() not center vertically within Grid
### Description
if i use TextCenterVertical(), Label is not center vertically (windows, android, i dont have ios so can't try it), but if i use standard code, the apps runs perfect.

### Steps to Reproduce

`public partial class TestPage : BasePage<TestViewModel>
{
    private enum Baris
	{
		Date,
        Line,
        Date2,
        Line2,
    };

	private enum Kolom
	{
		Kol1,
		Kol2
	}

    public TestPage(TestViewModel vm) : base(vm, vm.Title)
    {
        Label lblDate1 = new Label().TextCenterVertical().Text("Date"),
            lblLine1 = new Label().TextCenterVertical().Text("Line");

        Label lblDate2 = new Label()
        {
            VerticalTextAlignment = TextAlignment.Center,
        }.Text("Date 2"),
            lblLine2 = new Label()
            {
                VerticalTextAlignment = TextAlignment.Center,
            }.Text("Line 2");

        DatePicker dtpDate1 = new(),
            dtpDate2 = new();

        Entry txtLine1 = new()
        {
            Keyboard = Keyboard.Numeric,
            MaxLength = 11
        },
        txtLine2 = new()
        {
            Keyboard = Keyboard.Numeric,
            MaxLength = 11
        };

        Grid grid = new Grid
        {
            Padding = new(20),
            ColumnSpacing = 20,
            RowSpacing = 20,

            RowDefinitions = Rows.Define(Star, Star, Star, Star),
            ColumnDefinitions = Columns.Define(Stars(.30), Stars(.70)),
        };

        grid.Add(lblDate1, (int)Kolom.Kol1, (int)Baris.Date);
        grid.Add(dtpDate1, (int)Kolom.Kol2, (int)Baris.Date);

        grid.Add(lblLine1, (int)Kolom.Kol1, (int)Baris.Line);
        grid.Add(txtLine1, (int)Kolom.Kol2, (int)Baris.Line);

        grid.Add(lblDate2, (int)Kolom.Kol1, (int)Baris.Date2);
        grid.Add(dtpDate2, (int)Kolom.Kol2, (int)Baris.Date2);

        grid.Add(lblLine2, (int)Kolom.Kol1, (int)Baris.Line2);
        grid.Add(txtLine2, (int)Kolom.Kol2, (int)Baris.Line2);

        ScrollView scrollView = new()
        {
            Content = grid
        };

        Content = scrollView;
    }
}`

### Expected Behavior

test1

u can see the image, the 2 label above is not center vertically, but 2 label below is center vertically with the Entry Component

### Basic Information

- Version with issue:
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### PR DESCRIPTION
**Pull request by brminnick**
### Description of Change ###

 

This PR removes the usage of `TextAlignmentElement.HorizontalTextAlignmentProperty` and `TextAlignmentElement.VerticalTextAlignmentProperty`, both of which are no longer used by `Label`.

Because `ITextAlignment` implements exclusively read-only Properties, we cannot directly set `ITextAlignment.VerticalTextAlignment` or `ITextAlignment.VerticalTextAlignment`.

The solution is to create an extension method for each .NET MAUI Control that implements `Microsoft.Maui.ITextAlignment`, then set `ITextAlignment.HorizontalTextAlignment` and `ITextAlignment.VerticalTextAlignment` directly. 

Here is an example for `Editor`:

```cs
public static Editor TextCenterHorizontal(this Editor textAlignmentControl)
{
    textAlignmentControl.HorizontalTextAlignment = TextAlignment.Center;
    return textAlignmentControl;
}
```

The following "new" files were copy/pasted from `CommunityToolkit.Maui.SourceGenerators`:
- `NamespaceSymbolExtensions`
- `SourceStringExtensions`

 ### Linked Issues ###
 

 - Fixes #97 

 ### PR Checklist ###
 
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests (if omitted, state reason in description)
 - [x] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui.Markup/blob/main/CONTRIBUTING.md#contributing-code---best-practices)

 ### Additional information ###
 
#### 1. Confirmed NuGet Package Includes Generated Files
I confirmed that `dotnet pack -c Release` includes the generated files by manually adding the locally generated NuGet Package to [HelloMauiMarkup](https://github.com/brminnick/HelloMauiMarkup):
Screen Shot 2022-09-05 at 9 40 18 PM

#### 2. Confirmed NuGet Package Does Not Include Dependency on `CommunityToolkit.Maui.Markup.SourceGenerators`

Screen Shot 2022-09-05 at 10 08 23 PM

#### 3. Enables Future Improvements

This PR also lays the groundwork to fix `.Text()` that require specific controls that could previously only be verified at runtime:
https://github.com/CommunityToolkit/Maui.Markup/blob/f88ba85b7bee7f12d38215ff2c4cb55052b0687c/src/CommunityToolkit.Maui.Markup/ElementExtensions.cs#L200-L233

### CODE REVIEW

File: src/CommunityToolkit.Maui.Markup.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs
Comment (pictos) [inline src/CommunityToolkit.Maui.Markup.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs:66]: Do we care if the switch lands in `default`? If don't let's just do nothing (:

File: src/CommunityToolkit.Maui.Markup.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs
Comment (pictos) [inline src/CommunityToolkit.Maui.Markup.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs:43]:
```suggestion
		StringBuilder result = new("<");
```

File: src/CommunityToolkit.Maui.Markup.SourceGenerators/SourceGenerators/TextAlignmentExtensionsGenerator.cs
Comment (pictos) [inline src/CommunityToolkit.Maui.Markup.SourceGenerators/SourceGenerators/TextAlignmentExtensionsGenerator.cs:43]: **Note for the future**

If the next SG that we implement uses the same types as this list, we should create a `record` for that. IMO that will be cleaner. If we don't use the same members again we can leave it as it's.

File: src/CommunityToolkit.Maui.Markup.UnitTests/CommunityToolkit.Maui.Markup.UnitTests.csproj
Comment (pictos) [inline src/CommunityToolkit.Maui.Markup.UnitTests/CommunityToolkit.Maui.Markup.UnitTests.csproj:8]: Maybe we want to comment on this line. It can cause the LongPath error. This is useful for Mac env. where you can't inspect the generated code.

**Review by pictos — APPROVED** [review]
@brminnick great to work! LGTM!

My comments don't block the merge.

File: src/CommunityToolkit.Maui.Markup.UnitTests/CommunityToolkit.Maui.Markup.UnitTests.csproj
Comment (brminnick) [reply src/CommunityToolkit.Maui.Markup.UnitTests/CommunityToolkit.Maui.Markup.UnitTests.csproj:8]: I added this to match `CommunityToolkit.Maui.UnitTests.csproj`:

https://github.com/CommunityToolkit/Maui/blob/263ce321e6ecfd59480c7fc9229d7d1fa5d2bfc1/src/CommunityToolkit.Maui.UnitTests/CommunityToolkit.Maui.UnitTests.csproj#L7-L8

I'll leave it in for now. And if we get complaints and/or remove it from `CommunityToolkit.Maui.UnitTests.csproj`, we can remove it here as well.

File: src/CommunityToolkit.Maui.Markup.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs
Comment (brminnick) [reply src/CommunityToolkit.Maui.Markup.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs:66]: Unsure! I blindly copy/pasted this from your code in `CommunityToolkit.Maui.SourceGenerators` 😄

**Comment by brminnick** [comment]
Re-requesting a review from @pictos after merging the commit suggestion dismissed his approval

**Review by pictos — APPROVED** [review]

**Pull request merged**
